rasterio/rio/pyramid.py

```diff
@@ -0,0 +1,77 @@
+# coding: utf-8
+"""Manage overviews of a dataset."""
+
+import logging
+
+import click
+
+import rasterio
+from rasterio.enums import Resampling
+
+from . import options
+
+
+def build_handler(ctx, param, value):
+    if value:
+        try:
+            if '^' in value:
+                base, exp_range = value.split('^')
+                exp_min, exp_max = (int(v) for v in exp_range.split('..'))
+                value = [pow(int(base), k) for k in range(exp_min, exp_max+1)]
+            else:
+                value = [int(v) for v in value.split(',')]
+        except Exception as exc:
+            raise click.BadParameter(u"must match 'n,n,n,…' or 'n^n..n'.")
+    return value
+
+
+@click.command('pyramid', short_help="Construct overviews in an existing dataset.")
+@options.file_in_arg
+@click.option('--build', callback=build_handler, metavar=u"f1,f2,…|b^min..max",
+              help="A sequence of decimation factors specied as "
```

Comment:
@brendan-ward I'd like to standardize the CLI as `rio VERB` and I wasn't able to summon up a neat overviewing verb.

Here's an interesting convergence of "over view" and "pyramid". I'm a little leery of associating Rasterio with Freemasonry, though :fearful: 

![](https://upload.wikimedia.org/wikipedia/commons/3/3d/Dollarnote_siegel_hq.jpg)